### PR TITLE
024 registration system

### PR DESCRIPTION
**Pull request by joshuasy10**

### CODE REVIEW

File: src/main/java/org/kainos/ea/dao/AuthDao.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    public boolean isEmailTaken(String email) throws SQLException, DatabaseConnectionException {
+        Connection c = databaseConnector.getConnection();
+        Statement st = c.createStatement();
+        ResultSet rs = st.executeQuery("SELECT email FROM `users` " +
+                "WHERE email = '" + email + "'");
+
+        return rs.next();
+    }
+
+    public void deleteUser(String email) throws SQLException, DatabaseConnectionException {
```
Comment (KeelanScott) [inline src/main/java/org/kainos/ea/dao/AuthDao.java:119]: Is this necessary for this ticket?

Comment (joshuasy10) [reply src/main/java/org/kainos/ea/dao/AuthDao.java:119]: yes

File: src/test/java/integration/controller/AuthControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        Response responseLogin = APP.client().target("http://localhost:8080/api/login")
+                .request()
+                .post(Entity.entity(invalidLogin, MediaType.APPLICATION_JSON_TYPE));
+
+        Assertions.assertEquals(400, responseLogin.getStatus());
+    }
+
+    @Test
+    void register_shouldReturn200_whenSuccessfulRegister() throws DatabaseConnectionException, SQLException {
+        Login login = new Login(
```
Comment (KeelanScott) [inline src/test/java/integration/controller/AuthControllerIntegrationTest.java:65]: Shoud be an invalid login at the top to use instead

Comment (joshuasy10) [reply src/test/java/integration/controller/AuthControllerIntegrationTest.java:65]: im doing this as i want to be careful when deleting it after, you have added those env logins, i assume u want it to still work after this test runs

**Review by KeelanScott — COMMENTED** [review]
Looks good. Great start. Very thorough

**Review by KeelanScott — APPROVED** [review]

File: src/main/java/org/kainos/ea/service/AuthService.java [hunk 3]
```diff
@@ -24,4 +30,16 @@ public String login(Login login) throws FailedToLoginException, FailedToGenerate
             throw new FailedToGenerateTokenException();
         }
     }
+
+    public String register(Login login) throws FailedToGenerateTokenException, FailedToRegisterException, InvalidLoginException {
+        try{
+            authValidator.isValidLogin(login);
+            if(authDao.register(login)){
+                return authDao.generateToken(login.getEmail());
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/service/AuthService.java:38]: remove this as register shouldn't log someone in. think about when you register on a website, they always make you log in after successful registration

**Review by shaunganley — APPROVED** [review]
great work. approved with 1 small comment

**Pull request closed without merging**